src/server.js

```diff
@@ -13,25 +22,31 @@ MeteorX.onReady = function(cb) {
 };
 
 MeteorX.Server = Meteor.server.constructor;
+
+installMongoDetector(MeteorX);
+
 exposeLivedata(MeteorX);
 
-// before using any other MeteorX apis we need to hijack Mongo related code
-// that'w what we are doing here.
-Meteor.startup(function() {
-  runWithAFiber(function() {
+function initSync() {
+  runWithAFiber(() => {
     exposeMongoLivedata(MeteorX);
   });
 
-  MeteorX._readyCallbacks.forEach(function(fn) {
-    runWithAFiber(fn);
-  });
+  MeteorX._readyCallbacks.map(runWithAFiber);
   MeteorX._ready = true;
-});
+}
+
+async function initAsync() {
+  await exposeMongoAsync(MeteorX);
```

Comment:
This is called twice: once in `installMongoDetector`, and again here. This second call doesn't seem to do anything.

The goal was to only expose the classes from the mongo package if the app connected to MongoDB.

The current solution has two issues:
1. It fails to detect it if all collections are created before meteorx loads (the app doesn't use any collections, but has packages that do. This could happen in apps that have a package based architecture).
2. there is no guarantee that a collection will be created before `initAsync` is called - it could happen later, after the ready callbacks are run.

For the first one, there is no great option - we are going to have false negatives or false positives. I'm not sure what the right solution is. Another idea is to check if the app is using the mongo package, and the MONGO_URL env var is defined and then expose mongo. The main downside is if for some reason an app doesn't create any mongo collections, but loads the mongo package and has an old and broken MONGO_URL, it would cause the app to crash. Meteorx already has that issue and so far we haven't received any reports about it.

For the second issue, we might want to add a separate callback for mongo or some other way or something similar, or use the previous suggestion which would cause it to always be exposed before the ready callbacks are run.

However, we want to first release a version without breaking changes, so it should always expose mongo like meteorx currently does. This would be relevant for the first version 3 release.

Reply:
I removed the optional mongo behavior, saved this comment for later.